rotkehlchen/exchanges/htx.py

```diff
@@ -188,13 +223,131 @@ def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
 
         return returned_balances, ''
 
+    def _paginated_query(
+            self,
+            endpoint: Literal['/v1/query/deposit-withdraw', '/v1/order/matchresults'],
+            options: dict[str, Any],
+            start_ts: Timestamp,
+    ) -> list[dict[str, Any]]:
+        """
+        Query endpoints that can return paginated data. At the moment this is used for asset
+        movements. For asset movements we use cursor pagination since for any time range
+        the api will return a valid response and we don't know when to stop querying until
+        we find a lower updated_at than the start_ts(starting timestamp). The cursor pagination
+        uses the id of the last object in the last query to return the results of the next page.
+
+        May raise:
+        - RemoteError: if the network request fails
+        """
+        result = []
+        # copy since we are modifying the dict and could affect other queries using
+        # the same options
+        query_options = options.copy()
+        while True:
+            output = self._query(absolute_path=endpoint, options=options)
+            if len(output) == 0:
+                break
+
+            result.extend(output)
+            try:
+                query_options['from'] = output[-1]['id'] + 1  # from is inclusive, so we want after
+                updated_at = output[-1]['updated-at']  # the results are in descending order
```

Comment:
I just noticed. Is it updated-at what you need here? I believe you want a `created-at` since updated-at can be something much later